### PR TITLE
Api assignment2

### PR DESCRIPTION
**Pull request by waghmaresuyog**

### CODE REVIEW

File: API_Auto/src/test/java/postAllProduct/AllProduct.java [hunk 1]
```diff
@@ -1,4 +1,5 @@
-package APIAssignments;
+package postAllProduct;
+
```
Comment (anushreeSonak) [inline API_Auto/src/test/java/postAllProduct/AllProduct.java]: Please remove extra space

File: API_Auto/src/test/java/postAllProduct/AllProduct.java [hunk 2]
```diff
@@ -26,21 +27,19 @@ public AllProduct() {
...
-                .then().extract().response();
+        Response response = RestAssured.post(url).then().extract().response();
         JsonPath jsonResponse = new JsonPath(response.asString());
         ResponseBody body = response.getBody(); //use to print response body
         log.info("Response Body is: " + body.asString());
-        log.info("The response message is : " + jsonResponse.get("message"));
-        log.info("The response code is : " + jsonResponse.get("responseCode"));
+        Assert.assertEquals(response.getStatusCode(), 200);
+        log.info("the Status code is : " + response.getStatusCode());
+
```
Comment (anushreeSonak) [inline API_Auto/src/test/java/postAllProduct/AllProduct.java]: same here remove extra space

Comment (waghmaresuyog) [reply API_Auto/src/test/java/postAllProduct/AllProduct.java]: done

File: API_Auto/src/test/java/postAllProduct/AllProduct.java (hunk 1 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/postAllProduct/AllProduct.java]: done

**Review by anushreeSonak — APPROVED** [review]

File: API_Auto/src/test/java/postAllProduct/AllProduct.java [hunk 3]
```diff
@@ -26,21 +26,18 @@ public AllProduct() {
...
     public void checkResponse() {
-        Response response = RestAssured.post(url)
-                .then().extract().response();
+        Response response = RestAssured.post(url).then().extract().response();
         JsonPath jsonResponse = new JsonPath(response.asString());
         ResponseBody body = response.getBody(); //use to print response body
         log.info("Response Body is: " + body.asString());
-        log.info("The response message is : " + jsonResponse.get("message"));
-        log.info("The response code is : " + jsonResponse.get("responseCode"));
+        Assert.assertEquals(response.getStatusCode(), 200);
```
Comment (amitsahu8) [inline API_Auto/src/test/java/postAllProduct/AllProduct.java]: Resolve the conflicts and remove the unnecessary logs

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/postAllProduct/AllProduct.java (hunk 3 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/postAllProduct/AllProduct.java]: done

File: API_Auto/src/test/java/postAllProduct/AllProduct.java [hunk 4]
```diff
@@ -26,22 +26,18 @@ public AllProduct() {
...
-        System.out.println("The response code is : " + response.getStatusCode());
         log.info("The response code is : " + response.getStatusCode());
         
     }
 
     @Test(priority = 2)
     public void checkResponse() {
-        Response response = RestAssured.post(url)
-                .then().extract().response();
+        Response response = RestAssured.post(url).then().extract().response();
```
Comment (amitsahu8) [inline API_Auto/src/test/java/postAllProduct/AllProduct.java]: Reuse the existing code that has been implemented in API Assignment 1

**Review by amitsahu8 — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/postAllProduct/AllProduct.java (hunk 4 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/postAllProduct/AllProduct.java]: done

File: API_Auto/src/test/java/endPoint/EndPoint.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package endPoint;
+
+import helper.UrlReader;
+
+import java.io.IOException;
+
+public class EndPoint extends UrlReader {
+    //This is End url read from properties file and return the end url
+    public static String productEndURL() throws IOException {
+        return getProperties().getProperty("endUrl");
```
Comment (amitsahu8) [inline API_Auto/src/test/java/endPoint/EndPoint.java]: hardcode the endpoint

File: API_Auto/src/test/java/endPoint/ServiceUrl.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+        String getFinalUrl = productBaseUrl() + productEndURL();
+        return getFinalUrl;
+    }
+
+    // combine base url and end point store into finalUrl and return it
+    public static String giveUrl() {
+        String finalUrl = productBaseUrl() + productEndURL();
+        return finalUrl;
+    }
```
Comment (prashantkawa) [inline API_Auto/src/test/java/endPoint/ServiceUrl.java]: why 2 methods for same logic

File: API_Auto/src/test/java/configReader/UrlReader.java [hunk 7]
```diff
@@ -1,22 +1,16 @@
...
+package configReader;
 
 import java.io.FileInputStream;
 import java.io.IOException;
 import java.util.Properties;
 
 public class UrlReader {
-
+    //Read Properties file
     public static Properties getProperties() throws IOException {
```
Comment (prashantkawa) [inline API_Auto/src/test/java/configReader/UrlReader.java]: Use try catch block to handle the exceptions

File: API_Auto/src/test/java/testPostProductList/AllProductTestCases.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+import io.restassured.path.json.JsonPath;
+import io.restassured.response.ResponseBody;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class AllProductTestCases extends ProductsListMethod {
+    private static final Logger log = LogManager.getLogger("AllProduct.class");
```
Comment (prashantkawa) [inline API_Auto/src/test/java/testPostProductList/AllProductTestCases.java:13]: add space here

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/endPoint/ServiceUrl.java (hunk 6 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/endPoint/ServiceUrl.java]: done

File: API_Auto/src/test/java/configReader/UrlReader.java (hunk 7 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/configReader/UrlReader.java]: Done

File: API_Auto/src/test/java/configReader/UrlReader.java [hunk 9]
```diff
@@ -1,16 +1,27 @@
...
-        properties.load(file);
+        try {
+            file = new FileInputStream(userDirectory + "/Config.properties");
+
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException(e);
+        }
+        try {
+            properties.load(file);
+        } catch (IOException e) {
```
Comment (prashantkawa) [inline API_Auto/src/test/java/configReader/UrlReader.java]: why multiple try, when you can use multiple catch with single try block

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/configReader/UrlReader.java (hunk 9 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/configReader/UrlReader.java]: Done

File: API_Auto/src/test/java/testPostProductList/AllProductTestCases.java (hunk 8 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/testPostProductList/AllProductTestCases.java:13]: Done

File: API_Auto/src/test/java/configReader/UrlReader.java [hunk 10]
```diff
@@ -1,16 +1,27 @@
...
+            throw new RuntimeException(e);
+        }
```
Comment (prashantkawa) [inline API_Auto/src/test/java/configReader/UrlReader.java]: remove extra try block

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/configReader/UrlReader.java (hunk 10 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/configReader/UrlReader.java]: done

File: API_Auto/src/test/java/configReader/Config.properties [hunk 11]
```diff
@@ -0,0 +1 @@
+UrlBase=https://automationexercise.com/api/
```
Comment (prashantkawa) [inline API_Auto/src/test/java/configReader/Config.properties]: Config.properties file should be in config package

File: API_Auto/src/test/java/base/CrudOperation.java [hunk 12]
```diff
@@ -0,0 +1,22 @@
+package base;
+
+import io.restassured.RestAssured;
+import io.restassured.response.Response;
+
+import static base.FinalServiceUrl.*;
+
+public class CrudOperation extends EndPointServiceUrl {
```
Comment (prashantkawa) [inline API_Auto/src/test/java/base/CrudOperation.java]: No need to extend EndPointServiceUrl class if not reusing the base class methods

**Review by prashantkawa — CHANGES_REQUESTED** [review]

File: API_Auto/src/test/java/base/CrudOperation.java (hunk 12 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/base/CrudOperation.java]: done

File: API_Auto/src/test/java/configReader/Config.properties (hunk 11 above)
Comment (waghmaresuyog) [reply API_Auto/src/test/java/configReader/Config.properties]: done

**Review by prashantkawa — APPROVED** [review]
Approving PR

**Review by amitsahu8 — APPROVED** [review]

**Pull request merged**